tracker/dmlc_tracker/tracker.py

```diff
@@ -407,12 +407,17 @@ def get_host_ip(hostIP=None):
     return hostIP
 
 
-def submit(nworker, nserver, fun_submit, hostIP='auto', pscmd=None):
+def submit(nworker, nserver, fun_submit, hostIP='auto', pscmd=None, addnl_envs=None):
     if nserver == 0:
         pscmd = None
 
     envs = {'DMLC_NUM_WORKER' : nworker,
             'DMLC_NUM_SERVER' : nserver}
+
+    if addnl_envs:
+        for k in addnl_envs.split(","):
+            envs[k] = os.environ[k]
```

Comment:
If the key is already present in envs like DMLC_NUM_WORKER or DMLC_NUM_SERVER, shouldn't we skip it ?

Reply:
Well in that case, do you expect the user to have a different value for such environment variables on the host OS, and the value in envs gets changed? 
If they are the same value, it doesn't matter. If they can be different, I can add a check to prevent the user from making a mistake

Reply:
Yes I think a check would help here. If there is a env variable set and user has passed it to the script too, we should use the value passed to the script.